pallets/identity/src/claims.rs

```diff
@@ -481,6 +481,19 @@ impl<T: Config> Module<T> {
         // Sender has to be part of CDDProviders
         Self::ensure_authorized_cdd_provider(cdd_did)?;
 
+        let record = <DidRecords<T>>::get(cdd_did);
+        let cost = secondary_keys.iter().try_fold(
+            0usize,
+            |cost, auth| -> Result<usize, DispatchError> {
+                //Check permissions limits
+                Self::ensure_perms_length_limited(&auth.permissions)?;
+                Ok(cost.saturating_add(auth.permissions.complexity()))
+            },
+        )?;
+
+        // Check secondary keys limits
+        Self::ensure_secondary_keys_limited(&record, secondary_keys.len(), cost)?;
```

Comment:
The logic here looks strange to me. `record` comes from the DID of the CDD provider, so this code is currently checking that we can add the secondary keys to the CDD provider, not the *target*. What you should do instead is construct an empty `DidRecord` as `record` and pass that into `ensure_secondary_keys_limited`.

Reply:
Good catch.  I missed that.